internal/plugin/server.go

```diff
@@ -361,26 +374,24 @@ func (plugin *NvidiaDevicePlugin) getAllocateResponse(requestIds []string) (*plu
 // This includes per-resource pipe and log directories as well as a global daemon-specific shm
 // and assumes that an MPS control daemon has already been started.
 func (plugin NvidiaDevicePlugin) updateResponseForMPS(response *pluginapi.ContainerAllocateResponse) {
-	pipeDir := filepath.Join("/mps", string(plugin.rm.Resource()), "pipe")
-	response.Envs["CUDA_MPS_PIPE_DIRECTORY"] = pipeDir
+	// TODO: We should check that the deviceIDs are shared using MPS.
+	for k, v := range plugin.mpsDaemon.Envvars() {
+		response.Envs[k] = v
+	}
+
+	resourceName := plugin.rm.Resource()
 	response.Mounts = append(response.Mounts,
 		&pluginapi.Mount{
-			ContainerPath: pipeDir,
-			HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", pipeDir),
+			ContainerPath: plugin.mpsDaemon.PipeDir(),
+			HostPath:      plugin.mpsHostRoot.PipeDir(resourceName),
 		},
-	)
-	logDir := filepath.Join("/mps", string(plugin.rm.Resource()), "log")
-	response.Envs["CUDA_MPS_LOG_DIRECTORY"] = logDir
-	response.Mounts = append(response.Mounts,
 		&pluginapi.Mount{
-			ContainerPath: logDir,
-			HostPath:      filepath.Join("/var/lib/kubelet/device-plugins", logDir),
+			ContainerPath: plugin.mpsDaemon.PipeDir(),
+			HostPath:      plugin.mpsHostRoot.LogDir(resourceName),
 		},
-	)
-	response.Mounts = append(response.Mounts,
 		&pluginapi.Mount{
 			ContainerPath: "/dev/shm",
```

Comment:
Likewise have this be what is returned from a call to `plugin.mpsDaemon.ShmDir()`.